### PR TITLE
Buffer data instead of holding file reference

### LINKED ISSUE #30: Icon file not correctly closed
In follow up to PR #28 and issue #29 here is the simplest test to illustrate the problem.

I am still working through an alternative solution but I can't figure out how the  `os.Open(fname)` in icon.go is being closed without the `defer f.Close()`

```go
package goversioninfo

import (
	"log"
	"os"
	"runtime"
	"testing"
)

func TestTempIcon(t *testing.T) {
	icoPath := "tmp.ico"
	outPath := "resource.syso"
	vi := &VersionInfo{}
	vi.IconPath = icoPath

	f, _ := os.Create(icoPath)
	err := f.Close()
	if err != nil {
		log.Println("Unexpected error closing new file")
	}

	vi.Build()
	vi.Walk()
	err = vi.WriteSyso(outPath, runtime.GOARCH)
	if err != nil {
		log.Println("Unexpected error writing resource")
	}

	err = os.Remove(icoPath)
	if err != nil {
		t.Fatal("Error deleting temporary icon", err)
	}
}
```

### PR DESCRIPTION
**Pull request by andydotxyz**
The file handle was being retained by the io.SectionReader.
If we cache this data the file can be released and Close()d.

Fixes #30

### CODE REVIEW

**Comment by methane** [comment]
This looks very redundant.
When you buffering, no need to use SectionReader.
Both of two SectionReader is totally useless.

File: icon.go [hunk 1]
```diff
@@ -81,11 +82,25 @@ func addIcon(coff *coff.Coff, fname string, newID <-chan uint16) error {
...
+
+func bufferIcon(read *io.SectionReader) (*io.SectionReader, error) {
+	size := read.Size()
+	data := make([]byte, size)
+	_, err := io.ReadFull(read, data)
+	if err != nil {
+		return nil, err
+	}
+	return io.NewSectionReader(bytes.NewReader(data), 0, size), nil
+}
```
Comment (methane) [inline icon.go]:
```go
func bufferIcon(f *io.File, offset int64, size int) (coff.Sizer, error) {
    data := make([]byte, size)
    _, err := f.ReadAt(data, offset)
    if err != nil {
        return nil, err
    }
    return bytes.NewReader(data), nil
}
```

File: icon.go [hunk 2]
```diff
@@ -81,11 +82,25 @@ func addIcon(coff *coff.Coff, fname string, newID <-chan uint16) error {
 		for _, icon := range icons {
 			id := <-newID
 			r := io.NewSectionReader(f, int64(icon.ImageOffset), int64(icon.BytesInRes))
-			coff.AddResource(rtIcon, id, r)
+			buffered, err := bufferIcon(r)
+			if err != nil {
+				return err
+			}
+			coff.AddResource(rtIcon, id, buffered)
```
Comment (methane) [inline icon.go]:
```go
id := <-newID
// No io.NewSectionReader
buff, err := bufferIcon(f, int64(icon.ImageOffset), int(icon.BytesInRes))
if err != nil {
    return err
}
coff.AddResource(rtIcon, id, buff)
```

**Comment by andydotxyz** [comment]
Thanks @methane those are great suggestions, I will update the pull request shortly :)

**Pull request merged**